src/kerberos.c

```diff
@@ -211,14 +211,56 @@ static PyObject *authGSSClientClean(PyObject *self, PyObject *args)
     return Py_BuildValue("i", result);
 }
 
-static PyObject *authGSSClientStep(PyObject *self, PyObject *args)
+static PyObject *buildChannelBindingsStruct(PyObject *self, PyObject *args, PyObject* keywds)
+{
+    int initiator_addrtype = GSS_C_AF_UNSPEC;
+    int acceptor_addrtype = GSS_C_AF_UNSPEC;
+    int initiator_length = 0;
+    int acceptor_length = 0;
+    int application_length = 0;
+    const char *initiator_address = NULL;
+    const char *acceptor_address = NULL;
+    const char *application_data = NULL;
+    int result = 0;
+
+    PyObject *pychan_bindings = NULL;
+    struct gss_channel_bindings_struct *input_chan_bindings;
+	static char *kwlist[] = {"initiator_addrtype", "initiator_address", "accept_addrtype",
+	                        "acceptor_address", "application_data", NULL};
+
+	if (!PyArg_ParseTupleAndKeywords(args, keywds, "|is#is#s#", kwlist,
+	        &initiator_addrtype, &initiator_address, &initiator_length,
+	        &acceptor_addrtype, &acceptor_address, &acceptor_length,
+	        &application_data, &application_length)) {
+	    return NULL;
+	}
+
+	input_chan_bindings = (struct gss_channel_bindings_struct *) malloc(sizeof(struct gss_channel_bindings_struct));
+	pychan_bindings = PyCObject_FromVoidPtr(input_chan_bindings, free);
```

Comment:
Ok, just to say the words out loud for my own benefit: the differences between python 2 and 3 for instantiating python objects are abstracted by the macros near the top of src/kerberos.c, and the destructor function is conditionally defined based on the python version.